docs/training/apis-howto/api-pytorch-trainer-ug.rst

```diff
@@ -0,0 +1,354 @@
+   notebook).
+-  Run training scripts without needing to use an experiment configuration (YAML) file.
+-  Load previous saved checkpoints directly into your model.
+
+************
+ Objectives
+************
+
+After completing the steps in this guide, you will be able to do the following:
+
+-  Define a PyTorch Trial
+-  Initialize the PyTorch Trial and the Trainer
+-  Run Your Training Script Locally
+-  Submit Your Trial for Training On Cluster
+-  Load checkpoints
+
+***************
+ Prerequisites
+***************
+
+-  Access to a Determined cluster. If you have not yet installed Determined, refer to the
+   :ref:`installation instructions <install-cluster>`.
+
+-  The Determined CLI should be installed on your local machine. For installation instructions,
+   visit :ref:`Commands and Shells: Installation <install-cli>`. After installing the CLI, configure
+   it to connect to your Determined cluster by setting the ``DET_MASTER`` environment variable to
+   the hostname or IP address where Determined is running.
+
+********************************
+ Step 1: Define a PyTorch Trial
+********************************
+
+Start by defining a PyTorchTrial by instantiating the `Trial` and `TrialContext` objects. When using
+the PyTorch Trainer API, you do not have to initialize and wrap models and optimizers inside of the
+`Trial.__init__` method. If desired, pass a wrapped model to `Trial.__init__`.
+
+..
+   code::python
+
+   class MyPyTorchTrial(pytorch.PyTorchTrial):
+       def __init__(self, context: PyTorchTrialContext, hparams: Dict) -> None:
+           self.context = context
+           self.model = context.wrap_model(nn.Sequential(
+               nn.Linear(9216, 128),
+           ))
+           self.optimizer = context.wrap_optimizer(torch.optim.Adadelta(
+               self.model.parameters(), lr=hparams["lr"])
+           )
+
+       def train_batch(
+               self, batch: pytorch.TorchData, epoch_idx: int, batch_idx: int
+       ) -> Dict[str, torch.Tensor]:
+           ...
+           output = self.model(data)
+           loss = torch.nn.functional.nll_loss(output, labels)
+
+           self.context.backward(loss)
+           self.context.step_optimizer(self.optimizer)
+
+           return {"loss": loss}
+
+       def evaluate_batch(self, batch: pytorch.TorchData) -> Dict[str, Any]:
+           ...
+           return {"validation_loss": validation_loss, "accuracy": accuracy}
+
+       def build_training_data_loader(self) -> DataLoader:
+           ...
+           return DataLoader(train_set)
+
+       def build_validation_data_loader(self) -> DataLoader:
+           ...
+           return DataLoader(validation_set)
+
+******************************************************
+ Step 2: Initialize the PyTorch Trial and the Trainer
+******************************************************
+
+After defining the PyTorch Trial, initialize the trial and the trainer.
+
+   Example:
+
+   ..
+      code::python
```

Comment:
another accidental block comment

Reply:
or... actually the code in this code block looks like something out of a random internal docstring, not anything related to pytorch trainer

Reply:
not sure which code block this refers to
@azhou-determined  @rb-determined-ai